autoPyTorch/pipeline/components/preprocessing/base_preprocessing.py

```diff
@@ -23,7 +23,7 @@ def __init__(self) -> None:
         self.add_fit_requirements([
             FitRequirement('X_train',
                            (np.ndarray, pd.DataFrame, csr_matrix),
-                           user_defined=True, dataset_property=True)])
+                           user_defined=True, dataset_property=False)])
```

Comment:
Just double-checking here: setting this to False would mean that X_train shows as possible user input, is that what you need? Because if not then I'd leave it as True, then we can enforce that it only arrives at `fit` via the dataset class

Reply:
my idea was that we set dataset property to be true only for those requirements that need to be in the dataset properties dictionary. So, the requirements which have this flag(dataset_property) true are those that the method infer_dataset_properties will calculate and then return as a dict which can be added to the fit requirements either in the AutoML class or by the API. Also, X_train would either be loaded from the datamanager directly at early preprocessing, if it is small enough or in the data loader class, in my opinion, it need not be passed in the fit dictionary from the start. What do you think about this?

Reply:
I think that would be a clean solution (removing `X_train` from the requirements), also because in cross-validation these will differ anyways. One could also just have `X` and `split_id` as input.

About the flags, I think a `dataset_property=True` indicates that this is a quantity provided by the dataset (not necessarily in the `dataset_properties` dict, we might throw that argument out later anyways and just use its keys). 

`user_defined=True` indicates this is an input to be given when calling `pipeline.fit`. Conversely, `user_defined=False` indicates this is a `fit_requirement` produced by the pipeline itself (e.g. the initialized optimizer).

Reply:
I think yes, we could just have the split id in the fit dictionary. We don't need X also there as it is only needed in the early preprocessing node. So, if the dataset is small enough to be preprocessed we can get the train and val dataset directly at the preprocessing node with the backend and split id, and add this to the fit dictionary after preprocessing, then at the data loader stage if the dataset was preprocessed, we use the data present in the fit dictionary or else, we use the backend and split id to get the data there.